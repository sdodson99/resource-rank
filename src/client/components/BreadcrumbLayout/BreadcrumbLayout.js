import React from 'react';
import PropTypes from 'prop-types';
import styles from './BreadcrumbLayout.module.css';
import Layout from '../Layout/Layout';
import ActiveLink from '../ActiveLink/ActiveLink';

const BreadcrumbLayout = ({ children, breadcrumbs }) => {
  const breadcrumbItems = breadcrumbs.map(({ to, title }, index) => {
    const isLastBreadcrumb = breadcrumbs.length - 1 === index;

    return (
      <li key={to} className="flex">
        <ActiveLink
          href={to}
          className={styles.BreadcrumbItem}
          activeClassName={styles.ActiveBreadcrumbItem}
        >
          {title}
        </ActiveLink>

        {!isLastBreadcrumb && <div className="px-3">/</div>}
      </li>
    );
  });

  return (
    <div className={styles.BreadcrumbLayout} data-testid="BreadcrumbLayout">
      <Layout>
        <div className={`${styles.LayoutContent} content-container`}>
          <nav className={styles.BreadcrumbContent} aria-label="breadcrumb">
            <ol className="flex">{breadcrumbItems}</ol>
          </nav>
          <div className="mt-8">{children}</div>
        </div>
      </Layout>
    </div>
  );
};

BreadcrumbLayout.propTypes = {
  children: PropTypes.node,
  breadcrumbs: PropTypes.array,
};

BreadcrumbLayout.defaultProps = {
  breadcrumbs: [],
};

export default BreadcrumbLayout;
