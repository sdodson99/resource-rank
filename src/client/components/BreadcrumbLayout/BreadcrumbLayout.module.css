.BreadcrumbLayout {
}

.LayoutContent {
  @apply py-8;
}

.BreadcrumbContent {
  @apply bg-gray-50 rounded p-5 text-xl;
}

.BreadcrumbItem {
  @apply text-blue-700 underline;
}

.BreadcrumbItem.ActiveBreadcrumbItem {
  @apply text-gray-700 no-underline;
}
