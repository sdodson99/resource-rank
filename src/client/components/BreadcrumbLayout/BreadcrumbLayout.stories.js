import React from 'react';
import BreadcrumbLayout from './BreadcrumbLayout';

export default {
  component: BreadcrumbLayout,
  title: 'Components/BreadcrumbLayout',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <BreadcrumbLayout {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: <div>Hello world!</div>,
  breadcrumbs: [
    {
      to: '/',
      title: 'Home',
    },
    {
      to: '',
      title: 'Current',
    },
  ],
};

export const Single = Template.bind({});
Single.args = {
  children: <div>Hello world!</div>,
  breadcrumbs: [
    {
      to: '',
      title: 'Current',
    },
  ],
};
