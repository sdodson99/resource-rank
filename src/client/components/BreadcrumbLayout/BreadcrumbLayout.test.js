import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import BreadcrumbLayout from './BreadcrumbLayout';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<BreadcrumbLayout />', () => {
  test('it should mount', () => {
    render(<BreadcrumbLayout />);

    const breadcrumbLayout = screen.getByTestId('BreadcrumbLayout');

    expect(breadcrumbLayout).toBeInTheDocument();
  });

  describe('with multiple breadcrumbs', () => {
    const breadcrumbs = [
      {
        to: '/',
        title: 'Home',
      },
      {
        to: '/topics',
        title: 'topics',
      },
    ];

    it('should render correctly', () => {
      const page = createRenderer().render(
        <BreadcrumbLayout breadcrumbs={breadcrumbs} />
      );

      expect(page).toMatchSnapshot();
    });
  });

  describe('without breadcrumbs', () => {
    it('should render correctly', () => {
      const page = createRenderer().render(<BreadcrumbLayout />);

      expect(page).toMatchSnapshot();
    });
  });
});
