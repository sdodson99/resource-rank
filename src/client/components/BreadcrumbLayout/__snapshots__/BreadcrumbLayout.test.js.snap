// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadcrumbLayout /> with multiple breadcrumbs should render correctly 1`] = `
<div
  className="BreadcrumbLayout"
  data-testid="BreadcrumbLayout"
>
  <Layout>
    <div
      className="LayoutContent content-container"
    >
      <nav
        aria-label="breadcrumb"
        className="BreadcrumbContent"
      >
        <ol
          className="flex"
        >
          <li
            className="flex"
          >
            <ActiveLink
              activeClassName="ActiveBreadcrumbItem"
              className="BreadcrumbItem"
              href="/"
            >
              Home
            </ActiveLink>
            <div
              className="px-3"
            >
              /
            </div>
          </li>
          <li
            className="flex"
          >
            <ActiveLink
              activeClassName="ActiveBreadcrumbItem"
              className="BreadcrumbItem"
              href="/topics"
            >
              topics
            </ActiveLink>
          </li>
        </ol>
      </nav>
      <div
        className="mt-8"
      />
    </div>
  </Layout>
</div>
`;

exports[`<BreadcrumbLayout /> without breadcrumbs should render correctly 1`] = `
<div
  className="BreadcrumbLayout"
  data-testid="BreadcrumbLayout"
>
  <Layout>
    <div
      className="LayoutContent content-container"
    >
      <nav
        aria-label="breadcrumb"
        className="BreadcrumbContent"
      >
        <ol
          className="flex"
        />
      </nav>
      <div
        className="mt-8"
      />
    </div>
  </Layout>
</div>
`;
